### DISCUSSION TITLE
UTF8 support and/or wiki entries on U8G2 support

### DISCUSSION THREAD

**Discussion by ColinGarySmith** [moononournation/Arduino_GFX#588] [General]
I have loaded up a symbol font with u8g2, and it has fonts way up in the 2000+ range. Of course a char is limited to 0-254. How do i access those upper glyphs?

**Comment by moononournation** [comment]
you can refer to UTF8 example

**Comment by ColinGarySmith** [comment]
Ok so i dug some more and thank you for that.

So im looking at the example, which is using unicode, but, how would i specifically declare a unicode character?

I am using this font from u8g2:

https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_9x15_t_symbols.png

And want to use some of the symbols there. How would i "set" them in a gfx->println()?

For example, i want to try to use the degree C symbol.

Its 2103. How do i specify 2103 in the println?

**Reply by moononournation** [reply]
make the example run success first

**Comment by ColinGarySmith** [comment]
I actually got chatGPT to help. You could include this in your examples if you want to print custom code symbols from the u8g2 library.

in the main code include this:  (2103 is an example glyph)

uint32_t codepoint = 0x2103;
  char text[5]; 
  encodeUTF8(codepoint, text);
  gfx->print(text);
  
  
  which will create a UTF encoded char array that can be sent to the print function overload.
  
  This is the function that will convert a codepoint from one of the many many custom font faces that u8g2 supports so that any custom font glyph can be selected.
  
  
```
void encodeUTF8(uint32_t codepoint, char *output)
{
  if (codepoint <= 0x7F)
  { // 1-byte UTF-8
    output[0] = codepoint & 0x7F;
    output[1] = '\0';
  }
  else if (codepoint <= 0x7FF)
  { // 2-byte UTF-8
    output[0] = 0xC0 | ((codepoint >> 6) & 0x1F);
    output[1] = 0x80 | (codepoint & 0x3F);
    output[2] = '\0';
  }
  else if (codepoint <= 0xFFFF)
  { // 3-byte UTF-8
    output[0] = 0xE0 | ((codepoint >> 12) & 0x0F);
    output[1] = 0x80 | ((codepoint >> 6) & 0x3F);
    output[2] = 0x80 | (codepoint & 0x3F);
    output[3] = '\0';
  }
  else if (codepoint <= 0x10FFFF)
  { // 4-byte UTF-8
    output[0] = 0xF0 | ((codepoint >> 18) & 0x07);
    output[1] = 0x80 | ((codepoint >> 12) & 0x3F);
    output[2] = 0x80 | ((codepoint >> 6) & 0x3F);
    output[3] = 0x80 | (codepoint & 0x3F);
    output[4] = '\0';
  }
}
```
